### PR TITLE
feat: add support for setter prefix and suffix

### PR DESCRIPTION
**Pull request by Techassi**
This PR adds support to add prefixes and suffixes to setter methods.

```rust
#[derive(TypedBuilder)]
#[builder(field_defaults(setter(prefix = "with_", suffix = "_value")))]
struct Foo {
    x: i32,
    y: u32,
}

let foo = Foo::builder().with_x_value(1).with_y_value(2).build();
```

### CODE REVIEW

**Comment by mo8it** [comment]
May I ask why? It is just more to type repeatedly :/

**Comment by Techassi** [comment]
It is nice to have the option. For example I personally like to prefix my (manually implemented) builder methods with `with_`.

This is no breaking change and users don't have to set these values. If they are not set, the setter method is named like the field (current behavior). So really users can choose if they want to type more.

**Comment by mo8it** [comment]
I see your point, but I think that this subjective and adds complexity to the code. The whole crate and all its builder attributes have the purpose to type less.

I think that it is fine for a crate to be opionated to some degree.

But let's see what @idanarye thinks about it.

BTW: Always open an issue and wait for the response of the maintainers before starting with a PR. I was disappointed often enough by rejected PRs :)

**Comment by idanarye** [comment]
While I myself don't see the point in using `set_` or `with_` prefixes in a builder that does not grant public access to the internal fields, Typed Builder already allows customizing a lot of things so I don't see a point to get opinionated about this thing in particular. And while it does add complexity being a new feature and all, I think that complexity is very localized.

File: typed-builder-macro/src/struct_info.rs
Comment (idanarye) [inline typed-builder-macro/src/struct_info.rs:297]: Please change it to `setter_method_name`. While as a variable `method_name` makes sense inside the context of `field_impl`, it is pretty vague as a struct level method name. Especially when it's under `StructInfo`.

Speaking of which - why is this not under `FieldInfo`? And `FieldInfo` already has it's `name` inside it - there is no need to pass it twice.

File: typed-builder-macro/src/struct_info.rs
Comment (idanarye) [inline typed-builder-macro/src/struct_info.rs:303]: Wouldn't it make more sense to do this check in `SetterSettings::apply_meta`?

**Comment by idanarye** [comment]
Please add a test with a keyword identifier for a field. Typed Builder had a problem with those in the past (#47), and I suspect this new feature (especially the prefix) may choke on it as well.

**Comment by idanarye** [comment]
And please add an entry to the changelog.

File: typed-builder-macro/src/struct_info.rs
Comment (idanarye) [reply typed-builder-macro/src/struct_info.rs:303]: Also, this whole thing (converting an `syn::Expr` to a string) should probably be extracted to a function.

File: typed-builder-macro/src/struct_info.rs
Comment (Techassi) [reply typed-builder-macro/src/struct_info.rs]: Yes, that makes sense. Changed in cc1c5da

**Comment by Techassi** [comment]
> Also, I'm not sure how I feel about the prefix and the postfix automatically adding the underscore.

Removed the underscore in cc1c5da



> And please add an entry to the changelog.

Updated in fbae931

**Comment by Techassi** [comment]
> Please add a test with a keyword identifier for a field. Typed Builder had a problem with those in the past (#47), and I suspect this new feature (especially the prefix) may choke on it as well.

Added and fixed in ce48427

File: src/lib.rs
Comment (idanarye) [inline src/lib.rs:153]: The example here should say ``prefix = "with_"``. Similarly for the suffix.

File: typed-builder-macro/src/util.rs
Comment (idanarye) [inline typed-builder-macro/src/util.rs:92]: 1. 
    ```rust
    (name, name.len() < len)
    ```
2. `strip_raw_ident_prefix` is simple enough that it'd be simpler and more straightforward to duplicate its code here instead of 
    * Alternatively - reverse the roles, so that `strip_raw_ident_prefix` calls `strip_raw_ident_prefix_with_status` and ditches the status.
4. A boolean output "named" `status` is not very descriptive.
5. This entire thing is actually redundant - see my comment in `setter_method_name`.

File: typed-builder-macro/src/field_info.rs
Comment (idanarye) [inline typed-builder-macro/src/field_info.rs:80]: You don't actually need `was_stripped`. You only need to re-add the `r#` if both the prefix and the suffix are `None`, but in that case you can just return `self.name` (`.clone()`)

File: CHANGELOG.md
Comment (idanarye) [inline CHANGELOG.md:14]: Nit - "pre-" is confusing, and does not save much compared to just writing "prefixes".

File: typed-builder-macro/src/util.rs
Comment (Techassi) [reply typed-builder-macro/src/util.rs]: Removed in d7e98fc

File: typed-builder-macro/src/field_info.rs
Comment (idanarye) [inline typed-builder-macro/src/field_info.rs:88]: The point is that here you should return `self.name.clone()`, so that if the field name was `r#`'d the setter method name would be `r#`'d as well.

**Review by idanarye — CHANGES_REQUESTED** [review]

File: typed-builder-macro/src/field_info.rs
Comment (Techassi) [reply typed-builder-macro/src/field_info.rs:88]: Oh man, sorry. Totally missed that. Will change asap.

Comment (Techassi) [reply typed-builder-macro/src/field_info.rs]: Changed in c38d686

**Pull request merged**
